Fix a typo.

--- a/crypto.c
+++ b/crypto.c
@@ -271,7 +271,7 @@
 	/* Send AUTH command according to RFC 2554 */
 	send_remote_command(fd, "AUTH CRAM-MD5");
 	if (read_remote(fd, sizeof(buffer), buffer) != 3) {
-		syslog(LOG_DEBUG, "%s: smarthost authentification:"
+		syslog(LOG_DEBUG, "%s: smarthost authentication:"
 		       " AUTH cram-md5 not available: %s", it->queueid,
 		       neterr);
 		/* if cram-md5 is not available */
