Description: Cast correctly to uintmax_t when using %PRIxMAX.
Origin: other: http://svn.ringlet.net/svn/ringlet/mail/dma/
Forwarded: yes
Author: Peter Pentchev <roam@ringlet.net>
Last-Update: 2010-06-21

--- a/spool.c
+++ b/spool.c
@@ -92,7 +92,7 @@
 	 */
 	if (fstat(fd, &st) != 0)
 		goto fail;
-	if (asprintf(&queue->id, "%"PRIxMAX, st.st_ino) < 0)
+	if (asprintf(&queue->id, "%"PRIxMAX, (uintmax_t)st.st_ino) < 0)
 		goto fail;
 
 	queue->mailf = fdopen(fd, "r+");
